realm/realm-jni/src/io_realm_internal_TableView.cpp

```diff
@@ -1035,8 +999,7 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableView_nativeWhere(
         if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr))
             return 0;
 
-        Query query = TV(nativeViewPtr)->get_parent().where(TV(nativeViewPtr));
-        TableQuery* queryPtr = new TableQuery(query);
+        Query *queryPtr = new Query(std::move(TV(nativeViewPtr)->get_parent().where(TV(nativeViewPtr))));
```

Comment:
Well, `TableQuery` was also introduced so we could do a deep copy of the query. @beeender are you certain that the core's `Query` constructor does a deep copy?